### DISCUSSION TITLE
NUM_WARMUP_UPDATES multiplied by number of GPU's during multi-GPU training

### DISCUSSION THREAD

**Discussion by abhisirka2001** [SWivid/F5-TTS#1258] [Q&A]
Hi, 

I’m finetuning F5 on ~47k hours of multilingual data using 8 H100s. I set:

Base ckpt: https://huggingface.co/SWivid/F5-TTS/blob/main/F5TTS_Base/model_1200000.pt
NUM_WARMUP_UPDATES=20000

but during training the effective warmup appears to be 20k × 8 = 160k steps.

Is warmup counted per-GPU or globally? Is this scaling with world size expected, or should warmup be specified as global steps?

Also, do these params look reasonable for this scale?

BATCH_SIZE_PER_GPU=35000
LR=7e-5
EPOCHS=10

Any guidance on warmup/LR for large multi-GPU runs would be appreciated. Thanks!

**Comment by abhisirka2001** [comment]
created an issue
